packages/account-web-vault/src/web-user.ts

```diff
@@ -0,0 +1,250 @@
+                    'User cancelled login attempt by closing the QR code modal or an unexpected error occurred'
+                );
+
+                resolve(false)
+            }
+
+            const did = await account.did()
+            console.log(`Account connected with did: ${did}`)
+
+            WebUser.account = account
+            WebUser.context = context!
+            WebUser.did = did
+
+            const profile = await WebUser.getPublicProfile()
+
+            VeridaEvents.emit('connected', profile)
+            resolve(true)
+        })
+
+        return WebUser.connecting
+    }
+
+    public static async disconnect() {
+        await this.requireConnection()
+        await WebUser.context!.close()
+
+        WebUser.context = undefined
+        WebUser.account = undefined
+        WebUser.profile = undefined
+        WebUser.did = undefined
+        WebUser.connecting = new Promise((resolve) => { resolve(false) })
+
+        VeridaEvents.emit('disconnect')
+    }
+
+    /**
+     * Listen to an event on this user
+     * 
+     * @param eventName 
+     * @param cb 
+     */
+    public static on(eventName: string, cb: Function) {
+        // @ts-ignore
+        VeridaEvents.on(eventName, cb)
+    }
+
+    /**
+     * Send a message to a user's Verida Wallet
+     * 
+     * @param {*} did 
+     * @param {*} subject 
+     * @param {*} message 
+     * @param {*} linkUrl 
+     * @param {*} linkText 
+     */
+    static async sendMessage(did: string, message: WebUserMessage) {
+        const context = await WebUser.getContext()
+        const messaging = await context!.getMessaging()
+
+        const data = {
+            data: [{
+                subject: message.subject,
+                message: message.text,
+                link: message.link ? message.link : undefined
+            }]
+        }
+
+        const messageType = "inbox/type/message"
+        const config: MessageSendConfig = {
+            did,
+            recipientContextName: "Verida: Vault"
+        }
+
+        // Send the message across the network
+        await messaging.send(did, messageType, data, message.subject, config)
+    }
+
+    /**
+     * Is a user connected?
+     * Will auto-load from local storage if it exists.
+     * 
+     * @returns 
+     */
+    public static async isConnected() {
+        if (WebUser.did) {
+            return true
+        }
+
+        if (hasSession(WebUser.config.contextConfig.name)) {
+            const connected = await WebUser.connect()
+            return connected
+        }
+
+        return false
+    }
+
+    /**
+     * Throw an exception if a user isn't connected
+     */
+    public static async requireConnection() {
+        const isConnected = await WebUser.isConnected()
+        if (!isConnected) {
+            throw new Error('Not connected!')
+        }
+    }
+
+
+    // @todo
+    // public static async requestData() {}
+    // public static async sendData() {}
```

Comment:
A common use case for a connected account is to open a `datastore` for the current context.

It will be as simple as:

```ts
public static async openDatastore(schemaURL: string): Promise<Datastore> {
    const context = await WebUser.getContext();
    if (!context) {
        // Unlikely to happen if connected. See comment on narrowing the type
        throw new Error("Context undefined")
    }
    return await context.openDatastore(schemaURL);
}
```

Reply:
Done. Also added `openDatabase`.